### PR TITLE
Simpler `Jenkins.getComputers`

### PR DESCRIPTION
**Pull request by jglick**
While looking at a profiling snapshot of a controller running numerous builds with cloud agents, I noticed that `FilePath.toComputer` was called a lot, and most of its time was calling `SlaveComputer.getNode`.

getComputers

This was done solely to sort `MasterComputer` at the top. However that is silly because `MasterComputer.name` is the empty string, which would sort first anyway. Simplifying to just compare `Computer.name`, which should be cheaper.

Better still might be to introduce a separate (internal?) method to get an unsorted `Collection<? extends Computer>`, since most of the callers inside core (including `FilePath.toComputer`) do not actually need sorting, but I am hoping the new sort should be cheap enough it does not matter much.

### Testing done

Added test coverage, checking that it passed on the original version, and fails if the sorting in the new implementation is omitted or reversed.

### Proposed changelog entries

- Small optimization in computer list.

### CODE REVIEW

**Review by res0nance — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
